docs/usage/neo4j_container.md

```diff
@@ -0,0 +1,102 @@
+You can either use call `getHttpUrl()` or `getBoltUrl()` on the Neo4j container.
+`getHttpUrl()` gives you the HTTP-address of the transactional HTTP endpoint while `getBoltUrl()` is meant to be used with one of the [official Bolt drivers](https://neo4j.com/docs/developer-manual/preview/drivers/).
+On the JVM you would most likely use the [Java driver](https://github.com/neo4j/neo4j-java-driver).
+
+The following example uses the JUnit 5 extension `@Testcontainers` and demonstrates both the usage of the Java Driver and the REST endpoint:
+
+```java
+@Testcontainers
+public class ExampleTest {
+
+    @Container
+    private static Neo4jContainer neo4jContainer = new Neo4jContainer()
+        .withAdminPassword(null); // Disable password
+
+    @Test
+    void testSomethingUsingBolt() {
+
+        // Retrieve the Bolt URL from the container
+        String boltUrl = neo4jContainer.getBoltUrl();
+        try (
+            Driver driver = GraphDatabase.driver(boltUrl, AuthTokens.none());
+            Session session = driver.session()
+        ) {
+            long one = session.run("RETURN 1", Collections.emptyMap()).next().get(0).asLong();
+            assertThat(one, is(1L));
+        } catch (Exception e) {
+            fail(e.getMessage());
+        }
+    }
+
+    @Test
+    void testSomethingUsingHttp() throws IOException {
+
+        // Retrieve the HTTP URL from the container
+        String httpUrl = neo4jContainer.getHttpUrl();
+
+        URL url = new URL(httpUrl + "/db/data/transaction/commit");
+        HttpURLConnection con = (HttpURLConnection) url.openConnection();
+
+        con.setRequestMethod("POST");
+        con.setRequestProperty("Content-Type", "application/json");
+        con.setDoOutput(true);
+
+        try (Writer out = new OutputStreamWriter(con.getOutputStream())) {
+            out.write("{\"statements\":[{\"statement\":\"RETURN 1\"}]}");
+            out.flush();
+        }
+
+        assertThat(con.getResponseCode(), is(HttpURLConnection.HTTP_OK));
+        try (BufferedReader buffer = new BufferedReader(new InputStreamReader(con.getInputStream()))) {
+            String expectedResponse = 
+                "{\"results\":[{\"columns\":[\"1\"],\"data\":[{\"row\":[1],\"meta\":[null]}]}],\"errors\":[]}";
+            String response = buffer.lines().collect(Collectors.joining("\n"));
+            assertThat(response, is(expectedResponse));
+        }
+    }
+}
+```
+
+You are not limited to Unit tests and can of course use an instance of the Neo4j Testcontainer in vanilla Java code as well.
+
+
+## Choose your Neo4j license
+
+If you need the Neo4j enterprise license, you can declare your Neo4j container like this:
+
+```java
+@Testcontainers
+public class ExampleTest { 
+    @ClassRule
+    public static Neo4jContainer neo4jContainer = new Neo4jContainer()
+        .withEnterpriseEdition()
+        .acceptLicense();
```

Comment:
I quite like this approach of indicating licence acceptance in code, but we already have another mechanism that is used for the SQL Server image: [LicenceAcceptance](https://github.com/testcontainers/testcontainers-java/blob/master/core/src/main/java/org/testcontainers/utility/LicenseAcceptance.java). To avoid having two ways to do the same thing, please could we use that instead?

Reply:
Nice. I’ll change this.

Reply:
Done. Also solves the question wether Neo4j will always be free for development / CI.

I have added an example file for accepting the Neo4j license and adapted the test to run on the assumption whether it's there or not.